0000-hashed-timelock-agreements.md

```diff
@@ -0,0 +1,120 @@
+
+HTLAs thus work with ledgers that support hashlocks and timeouts (as HTLCs) and with ledgers that do not.
+
+For more details on HTLAs without ledger support see the sections below on [Simple Payment Channels](#simple-payment-channels) and [Trustlines](#trustlines).
+
+## Spectrum of HTLA Types
+
+The various types of HTLAs present a tradeoffs between complexity and risk. The more functionality a ledger provides, the less the users of that ledger need to trust one another.
+
+|  | Conditional Payment Channels (with HTLCs) | On-Ledger Holds/Escrow (using HTLCs) | Simple Payment Channels | Trustlines |
+|---|---|---|---|---|
+| **Ledger Support Required** | High | High | Medium | Low |
+| **Implementation Complexity** | High | Medium | Low | Low |
+| **Bilateral Risk** | Low | Low | Medium | High |
+
+### Conditional Payment Channels (with HTLCs)
+
+* **Ledger Features Required:** Payment Channels with HTLC Updates
+* **Non-Functional Ledger Requirements:** Fast*
+* **Money at Risk:** None
+* **Examples:** [Lightning-Style Channels](https://lightning.network)*
+
+With conditional payment channels, participants set up the channel by depositing funds in a shared, temporary account on the ledger. When a conditional transfer is prepared, the sender sends the recipient a signed update to the channel that includes a hashlock and timeout. The recipient may redeem the transfer amount if and only if they can present the hash preimage before the timeout. If the sender and recipient agree that the preimage was delivered before the timeout, they can exchange signed statements with the newly agreed upon channel balance. If there is a dispute, the recipient can present the last claim and the hashlock preimage to the ledger and the ledger will determine whether the preimage is valid and was submitted before the timeout.
+
+Using conditional payment channels, the sender and recipient can transact without any funds being at risk, because all disputes will be mediated by the ledger. This can be used as a mechanism to enable a greater volume of payments "through" a ledger than the ledger can natively support.
+
+&ast; **Note on ledger speed:** For a number of reasons, it is important that Interledger payments execute or fail quickly. With conditional payment channels, the recipient relies on the fact that they can redeem their funds even if there is a dispute with the sender by presenting their claims and the hashlock preimages to the ledger. However, this means that the transfer timeout must account for the ledger processing time, in case the recipient must resort to submitting the claims to the ledger. Thus conditional payment channels only work well with Interledger if the ledger can process claims in a few seconds or less.
+
```

Comment:
"conditional payment channels only work well with Interledger if the ledger can process claims in a few seconds or less" - that seems contradictory, knowing lightning uses them on top a very slow ledger.

I would turn the statement around, and say the payment timeout needs to be at least the ledger slowness, times the number of hops (so in the case of lightning, never less than 10 minutes * numHops). In practice, it's in everybody's interest to either accept or reject a payment much faster than this worst-case limit.